app/controllers/pages_controller.rb

```diff
@@ -2,13 +2,14 @@
 
 class PagesController < ApplicationController
   def home
-    @products = Product.where(query_params[:filter])
     @categories_select_params = Category.select_params
+    @brands_select_params = Brand.select_params
+    @products = Product.filtered(filter_params)
   end
-end
 
-private
+  private
 
-def query_params
-  params.permit(filter: [:category_id])
+  def filter_params
+    params.require(:filter).permit(:category_id, :brand_id) if params[:commit]
```

Comment:
ten if jest chyba niepotrzebny ;)

Reply:
No wałaśnie bez niego wywala 
```
 ActionController::ParameterMissing:
       param is missing or the value is empty: filter
```
bo ta forma jest w tej samej akcji(`home`) w kontrolerze 
``` 
def home
...
@products = Product.filtered(filter_params)
end
```
więc kiedy wchodzi się na pages#home pierwszy raz to parameter `:filter` jest nieobecny. 
Mógłbym dodać nową akcję w kontrolerze, dla samej formy ale to byłoby dużo więcej kodu i komplikacji niż ten jeden if.